test/unit/validation.test.ts

```diff
@@ -103,6 +104,15 @@ describe('Validation', () => {
                 expect(result.value).toEqual(login);
                 expect(result.error).toBeTruthy();
             });
+
+            it('should support a custom instance of Joi', () => {
+                validator = new Validator({
+                    joi: Joi,
+                });
+                const result = validator.validate(login);
+                expect(result.value).toEqual(login);
+                expect(result.error).toBe(null);
```

Comment:
With all these `'should support a custom instance of Joi'` tests, I don't believe we're actually testing which instance of joi it's using. Could we instead pass in a mocked version of joi that has a mocked validate method and check that it was called?

Something like...

```typescript
const joi = {
  validate: jest.fn()
} as Partial<typeof Joi> as typeof Joi;
validator = new Validator({ joi });
validator.validate(login);
expect(joi.validate).toHaveBeenCalled();
```

You might need to mock a return value for `validate` as well, and might be good to test what `joi.validate` is called with, but not essential imo.

Reply:
Great point, I'll sort it!